### PR TITLE
Neat 2.0.0 flex grid

### PR DESCRIPTION
**Pull request by wardpenney**
A basic flexbox grid. This is a simple stab at it to get the conversation going!

screen shot 2016-03-01 at 2 43 35 pm

### CODE REVIEW

**Comment by whmii** [comment]
holy code forks!

**Comment by wardpenney** [comment]
Batman!

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ Gemfile.lock
 node_modules
 public
 tmp
+npm-debug.log
```
Comment (tysongach) [inline .gitignore]: :abc:

File: contrib/base/_grids.scss [hunk 2]
```diff
@@ -1,4 +1,11 @@
 $grid--nested: (
   columns: 3,
   gutter: 0,
+  style: floated
```
Comment (tysongach) [inline contrib/base/_grids.scss]: Should `floated` be a default?

**Comment by tysongach** [comment]
I’m most curious about possible performance implications…

- https://benfrain.com/css-performance-test-flexbox-v-css-table-fight/
- https://jakearchibald.com/2014/dont-use-flexbox-for-page-layout/
- https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing?hl=en#use-flexbox-over-older-layout-models
- https://developers.google.com/web/updates/2013/10/Flexbox-layout-isn-t-slow?hl=en

Maybe it’s a non-issue.

**Comment by whmii** [comment]
I made a joke here earlier but one of my main concerns is that I consider the big code branches in current Neat to be one of its biggest weaknesses.

**Comment by wardpenney** [comment]
@whmii do you mean GitHub branches, or branching logic in methods? If it's the latter, there was a good Bike Shed where Shawn was talking about how he doesn't mind if a method has 1 branch / 1 decision to make. 2 or more, he factors into separate methods.

**Comment by whmii** [comment]
but if we can't decide which is better how is a user supposed to?

**Comment by tysongach** [comment]
@wardpenney I just rebased this branch off `neat-2.0.0`.

**Comment by katgironpe** [comment]
@tysongach is performance really an issue? I find Flexbox grid simple compared to float-based layouts. So yes, +1 for this.

**Comment by tysongach** [comment]
@katgironpe I don’t have a definitive answer, but it is something to think about. Flexbox is purpose-built more for component-level layout, and not really for page layout. For that, the W3 is working on CSS Grid.

**Comment by katgironpe** [comment]
@tysongach Interesting. Got my lazy self to google that. http://www.w3schools.com/w3css/w3css_grid.asp ... I am one of those who got convinced that Flexbox solves those common layout issues. https://philipwalton.github.io/solved-by-flexbox/ I never had problems with Neat for the past few years anyway. So I probably would not want such changes for Neat 2.0.

**Comment by whmii** [comment]
Based on some discussions that have been had, we will not be using flexbox for the grid system. This is mostly due to the fact that we would likely end up in a scenario where we are just renaming flexbox properties under the guise of an excedingly long list of mixin or options.

Happy to discuss the concept further, but if you need someone to hate, throw shade my way.

**Comment by whmii** [comment]
As a side note, you can always ad flex to a neat 2.0 row/container and it will still work as intended while giving you access to flexbox attributes like vertical centering, ect.

**Comment by tysongach** [comment]
@whmii If you add flexbox to a container with floated children, does that affect them?

**Comment by whmii** [comment]
@tysongach yep  because the margins & width is already defined you can basically add flex to your row and everything "just works". It's actually like super dope.

**Pull request closed without merging**